app/data/ContributionData.scala

```diff
@@ -14,7 +14,31 @@ import play.api.db.Database
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.Try
 
-class ContributionData(db: Database)(implicit ec: ExecutionContext) extends TagAwareLogger {
+trait ContributionData {
+  def insertPaymentHook(paymentHook: PaymentHook)(implicit tags: LoggingTags): EitherT[Future, String, PaymentHook]
+  def insertPaymentMetaData(pmd: ContributionMetaData)(implicit tags: LoggingTags): EitherT[Future, String, ContributionMetaData]
+  def saveContributor(contributor: Contributor)(implicit tags: LoggingTags): EitherT[Future, String, Contributor]
+}
+
+class CurrentAndNewContributionData(currentDB: ContributionDataInstance, newDB: ContributionDataInstance) extends ContributionData {
+
+  override def insertPaymentHook(paymentHook: PaymentHook)(implicit tags: LoggingTags): EitherT[Future, String, PaymentHook] = {
+    currentDB.insertPaymentHook(paymentHook)
+    newDB.insertPaymentHook(paymentHook)
```

Comment:
the `currentDB` result should probably be returned - just in case the result is being used anywhere - likewise for the other overridden methods in this class